From 5a9dea751835f9dbcb4f606c10d809259b943aa7 Mon Sep 17 00:00:00 2001
From: Rami Potinkara <rami.potinkara@qt.io>
Date: Fri, 26 Sep 2025 11:27:38 +0300
Subject: [PATCH 246/553] Android: SKIP tst_QRhiWidget::simple(OpenGL) on
 Android 16

This patch adds SKIP for:
- tst_QRhiWidget::simple(OpenGL) [crash]

This patch needs re-work later to find the root
cause and fix it.

Task-number: QTBUG-140627
Pick-to: 6.8
Change-Id: I68bb58660a9f9fbd9ccb4c1f20785a57974f9c22
Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
(cherry picked from commit d11bb09acff7df23b7396f8669d925fac7204428)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/widgets/widgets/qrhiwidget/tst_qrhiwidget.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tests/auto/widgets/widgets/qrhiwidget/tst_qrhiwidget.cpp b/tests/auto/widgets/widgets/qrhiwidget/tst_qrhiwidget.cpp
index da4fee72f65..35ca2f3c678 100644
--- a/tests/auto/widgets/widgets/qrhiwidget/tst_qrhiwidget.cpp
+++ b/tests/auto/widgets/widgets/qrhiwidget/tst_qrhiwidget.cpp
@@ -327,6 +327,13 @@ void tst_QRhiWidget::simple()
 {
     QFETCH(QRhiWidget::Api, api);
 
+#ifdef Q_OS_ANDROID
+    if ((QNativeInterface::QAndroidApplication::sdkVersion() == 36)
+        && (QRhiWidget::Api::OpenGL==api)) {
+        QSKIP("Fails and crashes on Android 16 (QTBUG-140627)");
+    }
+#endif
+
     SimpleRhiWidget *rhiWidget = new SimpleRhiWidget;
     rhiWidget->setApi(api);
     QSignalSpy frameSpy(rhiWidget, &QRhiWidget::frameSubmitted);
-- 
2.51.2

